src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/DebugTokenReplacer.cs

```diff
@@ -118,9 +121,11 @@ public DebugTokenReplacer(IUnconfiguredProjectCommonServices unconnfiguredServic
         /// This is here to support unit tests which can derive from this class and return their own 
         /// instance of IProjectReadAccess
         /// </summary>
-        protected virtual IProjectReadAccess AccessProject()
+        protected virtual async Task<IProjectReadAccess> AccessProject()
         {
-            return new ProjectReadAccessor(UnconfiguredServices.ProjectLockService, UnconfiguredServices.ActiveConfiguredProject);
+            var debuggingConfig = await ActiveDebugFramework.GetConfiguredProjectForActiveFrameworkAsync().ConfigureAwait(true);
```

Comment:
There is also a `ProjectConfig`. To make it unambiguous, rename this to `debuggingActiveConfiguredProject` or `debuggingConfiguredProject`?